packages/ember-glimmer/lib/utils/references.js

```diff
@@ -127,6 +127,11 @@ export class PropertyReference extends CachedReference { // jshint ignore:line
   }
 
   get(propertyKey) {
+    if (typeof propertyKey === 'number') {
+      // Allows us to get numeric keys
+      propertyKey = '' + propertyKey;
```

Comment:
I added the coercion here too, because of the assertion here: https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/property_get.js#L49